website/blog/2022-02-17-disggregated-coordinator.md

```diff
@@ -0,0 +1,152 @@
+
+#### Worker
+Each worker sends regular heartbeats with memory and cpu utilization to the resource managers. The resource managers track these metrics for the worker pool.
+
+## Query Execution Flow
+![QueryExecutionFlow](/img/blog/2022-02-17-disaggregated-coordinator/query-execution-flow.png)
+
+With the introduction of a resource manager, the query execution flow looks slightly different.
+
+- A query is submitted to one of the coordinators in the cluster.
+- The coordinator prepares the query for execution by parsing, analyzing and assigning it to a given resource group. 
+- A heartbeat is sent to each resource manager when the query is created by the coordinator.
+- The coordinator polls the resource manager at regular intervals to fetch cluster level resource group information.
+- The coordinator polls the resource manager to get active worker information. This information is used for query scheduling.
+- The rest of the query execution remains the same.
+
+
+## Memory Management
+![MemoryManagement](/img/blog/2022-02-17-disaggregated-coordinator/memory-management.png)
+
+The resource manager needs up to date information about memory and cpu utilization of the worker pool for resource group queuing. Currently, this information is periodically collected by the coordinator. In the disaggregated coordinator setup, resource managers receive query-level statistics from coordinator heartbeats, and memory pool information from worker heartbeats. This information is periodically polled by the coordinator to help make local decisions (i.e. queue/run a query, kill a query when the cluster is low in memory).
+
+## Resource Management
+![ResourceManagement](/img/blog/2022-02-17-disaggregated-coordinator/resource-management.png)
+
+The resource manager runs in multi-master mode. To support that, coordinators post query updates to all resource managers. The resource manager aggregates this information. The coordinator polls a resource manager to fetch up to date information about resource group usage in the cluster.
+
+## Resource Group Consistency Model
+
+Resource groups in a disaggregated coordinator setup are eventually consistent.  While this may lead to over-admission in certain scenarios, in practice this is mitigated by gating the resource group to only allow queries to run when certain freshness guarantees have been met (as opposed to the previous logic of checking every millisecond).  This may mean if the cluster’s resource managers are down, then queries may be queued in the coordinator’s resource groups.  This is to ensure coordinators don’t over-admit queries in the face of resource manager unavailability.
+
+The following flags can help one tune the cluster’s resource groups to the desired consistency:
```

Comment:
The flags should go into the Presto docs and be linked to from here :-) 
There shouldn't be anything to elaborate on that isn't already in the docs. Really if the blog post never existed the docs should be sufficient to explain what people need to know to enable and operate this feature (including why they would).